Exercise_5.py

```diff
@@ -1,10 +1,37 @@
 # Python program for implementation of Quicksort
 
 # This function is same in both iterative and recursive
+# TIME COMPLEXITY: O(n)
+# SPACE COMPLEXITY: O(1)
 def partition(arr, l, h):
-  #write your code here
-
+    #write your code here
+    greater_ind = l
+    pivot = arr[h]
+    for curr_ind in range(l, h):
+        if arr[curr_ind] < pivot:
+            arr[curr_ind], arr[greater_ind] = arr[greater_ind], arr[curr_ind]
+            greater_ind += 1
+    arr[greater_ind], arr[h] = arr[h], arr[greater_ind]
+    return greater_ind
 
+# TIME COMPLEXITY: O(n)
+# SPACE COMPLEXITY: O(n) - due to space of list_start_end - that has the number of calls to quicksortIterative.
```

Comment:
TC should be O(n^2) because of calling partition for every iteration of the loop.